Dishani-Created Scenario for TC0017 in Search.feature file

DishaniReddy opened the pull request:
Created scenario for below test case from QA Touch
Search - TC0017 - Verifying That In Autosuggestion Search Drop Down If Customer Entered Any Keyword Then In Search Box It Display Appropriate Option Which Enter In Search Box.

DishaniReddy committed:
Created Scenario for TC0017 in Search.feature file

sarangholey commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,9 @@ Feature: Search functionality
+		
+		@TC0017
+		Scenario: A User entered any keyword in search then in Search dropdown it should display appropriate options which is entered in the Search.
+	  Given I have browser opened and url is navigated
```
Plase keepa  proper indentation while writing the steps

DishaniReddy replied:
sure okay, but in eclipse, it's showing properly 
![image](https://user-images.githubusercontent.com/59105571/82973114-3e0b8380-9ff4-11ea-93d9-665560761f0d.png)

akashdktyagi replied:
ok

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,9 @@ Feature: Search functionality
 		When I search for product as "Dell"
 		And I click on any product in the Search Result
 		Then I am able to see product description and detail in new tab
-	  
```
No extra spaces between scnerios, only one line diff

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/resources/features/ui/Search.feature:
```diff
@@ -23,4 +23,9 @@ Feature: Search functionality
+		@TC0017
+		Scenario: A User entered any keyword in search then in Search dropdown it should display appropriate options which is entered in the Search.
+	  Given I have browser opened and url is navigated
+		When I search for product as "Dell"
```
you are not searching for Dell, you are just typing the keyword, so change the statement as when I type the keyword as "Dell"

DishaniReddy committed:
Updated Search.Stepdefs file and made required changes in
CmnPageObjects.java file by adding AutosuggestionSearchIn method. Also,
created a new JavascriptExecutor method in Interact.java file

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/runner/RunTest.java:
```diff
@@ -10,6 +10,7 @@
 
 		features="classpath:features",
 		glue="stepdefs",
+		tags="@TC0017",
```
Its fine what you do locally, but since this is a problem which is local to you and has not much relevance with the task, u should commit such changes. So now u have to revert this change. make the file as it is and then commit again.

akashdktyagi replied:
modify this.

DishaniReddy committed:
Finished implementation of the Assigned TestCase TC0017 in
SearchStepDefs.java file and also reverted the changes made earlier in
Interact.java file.

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/pageobjects/CmnPageObjects.java:
```diff
@@ -43,6 +47,27 @@ public void ClickOnSearchButton() {
+		setElement(search_text_box, Keys.DOWN);
+		
+		String text = getAttribute(search_text_box,"value");
+		System.out.println("Text is "+ text);	
```
we do not need these sys out statements. instead use scn object

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/pageobjects/CmnPageObjects.java:
```diff
@@ -43,6 +47,27 @@ public void ClickOnSearchButton() {
+			String  item = getAttribute(search_text_box,"value");
+			Thread.sleep(1000);
+			if(item.contains(productName)) {
+				System.out.println(item);
```
Use scn object.

akashdktyagi commented on Batch8CucumberSeleniumAPI/src/test/java/pageobjects/CmnPageObjects.java:
```diff
@@ -43,6 +47,27 @@ public void ClickOnSearchButton() {
+			if(item.contains(productName)) {
+				System.out.println(item);
+			}else {
+				System.out.println("Item Name does not match with the value entered in the text box!");
```
use scn ob ject.
Also, put an assert statement here. otherwise test case will always be displayed as pass.

DishaniReddy committed:
Moving Updated code where in I removed keys.down and made appropriate
changes

DishaniReddy committed:
Changed syso statement with scn.write

DishaniReddy committed:
Gave proper indentation in feature file

DishaniReddy committed:
Changed variable name and gave appropriate name

DishaniReddy committed:
As there was error & proper Output was not obtained, updated the code in
CmnpageObjects.java file

DishaniReddy committed:
Updated the code in CmnPageObject.java file using Assertion statement.
Used Soft Assert as of now
